lib/topologies/topology_base.js

```diff
@@ -390,9 +390,12 @@ class TopologyBase extends EventEmitter {
       this.s.sessionPool.endAllPooledSessions();
     }
 
-    this.s.coreTopology.destroy({
-      force: typeof forceClosed === 'boolean' ? forceClosed : false
-    });
+    this.s.coreTopology.destroy(
+      {
+        force: typeof forceClosed === 'boolean' ? forceClosed : false
+      },
+      callback
+    );
 
     // We need to wash out all stored processes
     if (forceClosed === true) {
```

Comment:
Should this be part of the callback after ` this.s.coreTopology.destroy(`?

Reply:
My thought was that if you are doing a force close, then flushing these immediately should be expected. If its not a force, then the async close will wait for all buffered operations to complete. Maybe this should go _before_ the call to `destroy`?

Reply:
Yeah, that might be the best route. does that affect the results of any of our tests?

Reply:
I just tested again with this change, no breakage